File: basics/rospy/simple_sub_py/src/simple_sub.py
#! /usr/bin/env python3

import sys
import rospy
from std_msgs.msg import String

def callback(msg):
    rospy.loginfo("I received: " + msg.data)

def main(args=None):
    rospy.init_node("simple_sub_node")

    sub = rospy.Subscriber("/talk", String, callback)

    rospy.spin()

if __name__ == "__main__":

    try:
        main(sys.argv)
    except rospy.ROSInterruptException:
        pass

File: basics/roscpp/simple_sub/src/simple_sub.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

void
callback(const std_msgs::Int32::ConstPtr & msg)
{
	ROS_INFO("I received: %d\n", msg->data);
}

int
main(int argc, char **argv)
{
	ros::init(argc, argv, "simple_sub_node");

	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/talk", 1, callback);

	ros::spin();

exit(EXIT_SUCCESS);
}
